pkg/controller/controller_instance.go

```diff
@@ -252,16 +252,46 @@ func (c *controller) reconcileInstance(instance *v1alpha1.Instance) error {
 
 	var parameters map[string]interface{}
 	if instance.Spec.Parameters != nil {
-		parameters, err = unmarshalParameters(instance.Spec.Parameters.Raw)
-		if err != nil {
-			s := fmt.Sprintf("Failed to unmarshal Instance parameters\n%s\n %s", instance.Spec.Parameters, err)
+		params := instance.Spec.Parameters
+		if params.SecretRef != nil {
+			parameters, err = fetchSecretParameters(c.kubeClient, instance.Namespace, params.SecretRef)
+			if err != nil {
+				s := fmt.Sprintf("Failed to fetch Instance secret parameters\n%s\n %s", instance.Spec.Parameters, err)
+				glog.Warning(s)
+				c.updateInstanceCondition(
+					instance,
+					v1alpha1.InstanceConditionReady,
+					v1alpha1.ConditionFalse,
+					errorWithParameters,
+					"Error fetching instance secret parameters. "+s,
```

Comment:
Minor thing but shouldn't we just use 's' here instead of repeating that we couldn't fetch the secret? This will appear as: "Error fetching instance secret parameters. Failed to fetch Instance secret parameters..."